src/main/java/com/sulphate/chatcolor2/listeners/ColorGUIListener.java

```diff
@@ -42,80 +52,84 @@ public void onEvent(InventoryClickEvent event) {
+                        activeModifier = Character.toString(item.getItemMeta().getDisplayName().charAt(3));
+                    }
+                }
+            }
 
-            // Get their colour.
             Player player = (Player) event.getWhoClicked();
-            String colourString = MainClass.getUtils().getColor(player.getUniqueId().toString());
-
-            String colour = colourString.substring(1, 2);
-            String modPart = colourString.substring(2);
-            String[] modifiers = modPart.split("&");
-
+            String playerUuid = player.getUniqueId().toString();
             String displayName = clickedItem.getItemMeta().getDisplayName();
             String firstLoreLine = clickedItem.getItemMeta().getLore().get(0);
 
             if (isColour) {
-                String colChar = Character.toString(displayName.charAt(1));
+                String newColor;
+                if (displayName.equals(CC2Utils.colouriseMessage("rainbow", CCStrings.rainbow, false))) {
+                    newColor = "rainbow";
+                }
+                else {
+                    newColor = Character.toString(displayName.charAt(1));
 
-                // For some reason, the white colour code doesn't display.
-                if (colChar.equals("h")) {
-                    colChar = "f";
+                    // For some reason, the white colour code doesn't display.
+                    // So we default to white if another color couldn't be assigned
+                    if (ChatColorCommand.getColor(newColor) == null) {
+                        newColor = "f";
+                    }
                 }
 
                 // Make sure it's not unavailable.
-                if (firstLoreLine.equals(CC2Utils.colourise("&cUnavailable"))) {
-                    player.sendMessage(CC2Utils.colourise(CCStrings.nocolorperms + colChar + colChar));
+                if (firstLoreLine.equals(CCStrings.guiunavailable)) {
+                    player.sendMessage(CCStrings.nocolorperms + displayName);
                 }
-                // Make sure it's not their current colour.
-                else if (!colChar.equals(colour)) {
-                    StringBuilder sb = new StringBuilder("&").append(colChar);
-
-                    for (String modifier : modifiers) {
-                        if (modifier.equals("")) continue;
-
-                        sb.append('&').append(modifier);
-                    }
-
-                    // Set their colour, re-open the GUI.
-                    String newColour = sb.toString();
-                    MainClass.getUtils().setColor(player.getUniqueId().toString(), newColour);
-
-                    player.sendMessage(CCStrings.setowncolor + CC2Utils.colourise(newColour) + CCStrings.colthis);
-                    openGUI(player); // Refreshes the GUI.
+                else {
+                    ChatColorCommand.setColorFromArgs(playerUuid, new String[]{ newColor, activeModifier });
```

Comment:
Yep, so as I said in my comment, it'd be great if you could account for multiple modifiers.

Reply:
Will do!

Reply:
Many thanks :)